tools/build-image-if-needed.sh

```diff
@@ -13,5 +13,5 @@ if git diff --exit-code remotes/origin/master...HEAD -- ${BUILD_DIRECTORY}; then
   echo "${BUILD_DIRECTORY} files not modified; no rebuild required"
 else
   echo "${BUILD_DIRECTORY} files modified; ${TAG_NAME} requires a rebuild"
-  docker build ${BUILD_DIRECTORY} -t ${TAG_NAME}
+  docker buildx build --push --platform linux/arm64/v8,linux/amd64 ${BUILD_DIRECTORY} -t ${TAG_NAME}
```

Comment:
`buildx` is not able to load multi-arch images into the local docker instance (see [here](https://docs.docker.com/engine/reference/commandline/buildx_build/#output)), so this builds and pushes in one step. I suspect that this might not be the desired behavior here, but I'm not sure what is. If this is just for a local development build, then maybe this file doesn't need to change at all.

Reply:
We use this script in CI to rebuild various images if need: https://github.com/PrairieLearn/PrairieLearn/blob/c0c0e638dda4e3f0cc8a921bdc9fd6ef6628279b/.github/workflows/main.yml#L15

I think we can leave this file unchanged, because we only really need to build for local usage here.